src/html/translate-directive-translation-extractor.ts

```diff
@@ -0,0 +1,150 @@
+    translationId = translateAttribute.value;
+  }
+
+  for (const attribute of element.attributes) {
+    handleAttributesWithTranslateExpressions(attribute, context);
+  }
+
+  if (translateDirective) {
+    const hasTranslateAttributes = handleTranslateAttributes(element, context);
+
+    const defaultTextAttribute = element.attributes.find(
+      attr => attr.name === "translate-default"
+    );
+
+    // TODO make safe
+    translationId =
+      translationId ||
+      (element.texts.length > 0 ? element.texts[0].text : null);
+
+    // <any translate='test'></any>
+    if (translationId) {
+      context.registerTranslation({
+        translationId,
+        defaultText: defaultTextAttribute
+          ? defaultTextAttribute.value
+          : undefined,
+        position: element.startPosition
+      });
+    } else if (!hasTranslateAttributes) {
+      // no translate-attr-* and the element has not specified a translation id, someone is using the directive incorrectly
+      context.emitSuppressableError(
+        "the element uses the translate directive but does not specify a translation id nor has any translated attributes (translate-attr-*). Specify a translation id or remove the translate-directive.",
+        element.startPosition
+      );
+    }
+  } else {
+    handleTexts(element, context);
+  }
+}
+
+type KeyedAttributes = { [name: string]: Attribute };
+
+function handleTranslateAttributes(
+  element: AngularElement,
+  context: HtmlTranslationExtractionContext
+) {
+  const keyedAttributes = element.attributes.reduce(
+    (obj, attribute) => {
+      obj[attribute.name] = attribute;
+      return obj;
+    },
+    {} as KeyedAttributes
+  );
+
+  // translate-attr-ATTR
+  const translateAttributes = element.attributes.filter(attr =>
+    translateAttributeRegex.test(attr.name)
+  );
+
+  for (const attribute of translateAttributes) {
+    const attributeName = attribute.name.substr("translate-attr-".length);
+    const defaultTextAttribute =
+      keyedAttributes[`translate-default-attr-${attributeName}`];
+
+    context.registerTranslation({
+      translationId: attribute.value,
+      defaultText: defaultTextAttribute
+        ? defaultTextAttribute.value
+        : undefined,
+      position: attribute.startPosition
+    });
+  }
+
+  return translateAttributes.length > 0;
+}
+
+// <any attr='{{ x | translate }}'></any>
+function handleAttributesWithTranslateExpressions(
+  attribute: Attribute,
+  context: HtmlTranslationExtractionContext
+) {
+  for (const expression of attribute.expressions) {
+    handleAngularExpression(expression, context, attribute.startPosition);
+  }
+}
+
+function handleAngularExpression(
+  expression: AngularExpressionMatch,
+  context: HtmlTranslationExtractionContext,
+  position: number
+) {
+  let translationId = expression.value;
+  // e.g { translate | var} instead of a string constant
+  if (!(/^".*"$/.test(translationId) || /^.*'$/.test(translationId))) {
```

Comment:
Missing single quote in second regex I reckon:
```suggestion
  if (!(/^".*"$/.test(translationId) || /^'.*'$/.test(translationId))) {
```

Reply:
Nice catch!